### PR TITLE
fix: Prevent the cart from resorting when the quantity is updated

### PR DESCRIPTION
**Pull request by lucasrocharodrigues**
## What is this contribution about?

Ao aumentar a quantidade no minicart o produto era removido e adicionado novamente, alterando a ordem dos produtos 

Esse PR corrige esse comportamento

### CODE REVIEW

File: wake/actions/cart/updateItemQuantity.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ const removeFromCart = (
     RemoveItemFromCartMutationVariables
   >({
     variables: {
-      input: { id: cartId, products: [{ ...props, quantity: 1e6 }] },
+      input: { id: cartId, products: [{ ...props }] },
```
Comment (mcandeia) [inline wake/actions/cart/updateItemQuantity.ts:51]:
```suggestion
      input: { id: cartId, products: [props] },
```

Comment (mcandeia) [reply wake/actions/cart/updateItemQuantity.ts:51]: why did you remove quantity?

File: wake/actions/cart/updateItemQuantity.ts [hunk 2]
```diff
@@ -65,18 +65,28 @@ const action = async (
...
+  const quantity = props.quantity - quantityItem
 
-  if (props.quantity > 0) {
-    data = await addToCart(props, cartId, ctx, headers);
+  let checkout
+
+  if (props.quantity > 0 && quantity > 0) {
+    checkout = await ctx.invoke.wake.actions.cart.addItem({...props, quantity})
+  } else {
+    const data = await removeFromCart({...props, quantity:quantity*-1}, cartId, ctx, headers);
```
Comment (mcandeia) [inline wake/actions/cart/updateItemQuantity.ts:78]:
```suggestion
    const data = await removeFromCart({...props, quantity:-quantity}, cartId, ctx, headers);
```

File: wake/actions/cart/updateItemQuantity.ts [hunk 3]
```diff
@@ -65,18 +65,28 @@ const action = async (
     throw new HttpError(400, "Missing cart cookie");
   }
 
-  let data = await removeFromCart(props, cartId, ctx, headers);
+  const cart = await ctx.invoke.wake.loaders.cart(props, req)
+  const item = cart.products?.find(item => item?.productVariantId === props.productVariantId)
+  const quantityItem = item?.quantity ?? 0
+  const quantity = props.quantity - quantityItem
```
Comment (mcandeia) [inline wake/actions/cart/updateItemQuantity.ts:71]: is this the left quantity? can you explain this calculation ? (add a js doc on top of it)

File: wake/actions/cart/updateItemQuantity.ts (hunk 1 above)
Comment (lucasrocharodrigues) [reply wake/actions/cart/updateItemQuantity.ts:51]: the quantity was used to remove all products imagining that there would be less than 1000000, which is not necessary considering that I pass the exact value that should be removed

File: wake/actions/cart/updateItemQuantity.ts (hunk 3 above)
Comment (lucasrocharodrigues) [reply wake/actions/cart/updateItemQuantity.ts:59]: yes. I add the doc, can you check?

File: wake/actions/cart/updateItemQuantity.ts [hunk 4]
```diff
@@ -65,18 +46,35 @@ const action = async (
...
+
+  let checkout
+
+  if (props.quantity > 0 && quantity > 0) {
+    checkout = await ctx.invoke.wake.actions.cart.addItem({...props, quantity})
+  } else {
+    const data = await removeFromCart({...props, quantity:-quantity}, cartId, ctx, headers);
```
Comment (mcandeia) [inline wake/actions/cart/updateItemQuantity.ts:66]: based on the formatting, I think you did miss a `deno task check` can you run it before get it merged?

**Review by mcandeia — APPROVED** [review]

**Pull request merged**